user_guide_src/source/database/query_builder/118.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+$data = [
+    [
+        'order'   => 48372,
+        'line'    => 3,
+        'product' => 'Keyboard',
+        'qty'     => 1,
+    ],
+    [
+        'order'   => 48372,
+        'line'    => 4,
+        'product' => 'Mouse',
+        'qty'     => 1,
+    ],
+    [
+        'order'   => 48372,
+        'line'    => 5,
+        'product' => 'Monitor',
+        'qty'     => 2,
+    ],
+];
+
+$builder->setData($data, true, 'del')
+    ->onConstraint('order, line')
+    ->where('del.qty >', 1)
+    ->deleteBatch();
+
+/*
+ * MySQL Produces:
+ * DELETE `order_line` FROM `order_line`
+ * INNER JOIN (
+ * SELECT 48372 `order`, 3 `line`, 'Keyboard' `product`, 1 `qty` UNION ALL
+ * SELECT 48372 `order`, 4 `line`, 'Mouse'    `product`, 1 `qty` UNION ALL
+ * SELECT 48372 `order`, 5 `line`, 'Monitor'  `product`, 2 `qty`
```

Comment:
```suggestion
 * SELECT 3 `line`, 48372 `order`, 'Keyboard' `product`, 1 `qty` UNION ALL
 * SELECT 4 `line`, 48372 `order`, 'Mouse'    `product`, 1 `qty` UNION ALL
 * SELECT 5 `line`, 48372 `order`, 'Monitor'  `product`, 2 `qty`
```

Reply:
I'm not sure why you suggest this. the table name is `order_line`... so order + line. The pk would have order first and then line.

Reply:
That's the result in my local environment. MySQL 5.7.

Reply:
Oh ok.. that is because setData() sorts the keys alphabetically.

Reply:
So technically it will get reordered but for readability I like the way it reads.. can we leave it this way?

Reply:
Why does setData() sort the keys?
If the output SQL will be difficult to read, it seems better not to sort.

Reply:
I don't know. That is the way things were previously. I left things as the were when designing setData().

Reply:
Done